### ISSUE TITLE
[Feature Request] Support same collection with different EntityCollection

### ISSUE THREAD

**Issue by jeemercado** [Camberi/firecms#255]
First of all, great library!

Problem:
I have a collection"Posts". I wanted to have a separate `EntityCollectionView` or page for `allPosts` and has `initialFilters` for `pendingPosts`

Ideal scenario on the navigation page. I have both "All Posts" and "Pending Posts".

I tried creating a custom view 
```
const pendingPostsCollection = buildCollection<Partial<Post>>({
      initialFilter: filterPendingPosts,
      name: 'Pending Posts',
      path: 'posts',
      permissions: editOnlyPermission,
      schema: pendingPosts,
      textSearchEnabled: true,
    });

    const customViews: CMSView[] = [
      {
        name: 'Pending Posts',
        path: ['pendingPosts'],
        // This can be any React component
        view: <EntityCollectionView collection={pendingPostsCollection} path="posts" />,
      },
    ];
```

But this implementation requires me to have an existing path for "posts" in navigation or else it doesn't resolve any schema even tho I already passed in a collection (My guess is it's binded somewhere on navigation). 

However, if I do put a path for "posts" which is the "All Posts" EntityCollection. It resolves to the existing an existing schema "All Posts" EntityCollection and doesn't work as intended since my `initialFilters` and `customized pendingPostsSchema` isn't there.

**Comment by ramespra** [comment]
I also have this problem

**Comment by jeemercado** [comment]
> 

Yes it would work for as long as the path (reference path) shouldn't resolve to the same collection. I did some digging. I'm not entirely sure how this is used (I didn't dig further enough) but on `EntityCollectionView.tsx` it seems like `collectionResolver`, `schemaResolver`, and `collection` is coupled with the path (reference path)

```
const collectionResolver = navigationContext.getCollectionResolver<M>(path);
    if (!collectionResolver) {
        throw Error(`Couldn't find the corresponding collection view for the path: ${path}`);
    }
  ...
    const collection: EntityCollection<M> = collectionResolver ?? baseCollection;

    const { schemaResolver } = collectionResolver;
```

Basically treat them as separate collections even if they belong to the same reference path.

**Comment by fgatti675** [comment]
Hi @jeemercado 
We have big changes regarding how collections are resolved internally for version 2.0 and will keep this feature in mind.
Yeah, right now the path acts as an id so it is hard to change this behaviour at this point.
Thank you for the feedback!

**Comment by fgatti675** [comment]
This feature will be available in version 2.0.0 since we hace introduced collection aliases, allowing developers to have multiple collections in the same path

**Comment by fgatti675** [comment]
This is now possible in version 2.0 using `alias` when defining your collection
